k8s/k8s.go

```diff
@@ -0,0 +1,206 @@
+		if err = uu.Write(filepath.Join(uu.VulnListDir(), "upstream", fmt.Sprintf("%s.json", cve.ID)), cve); err != nil {
+			return xerrors.Errorf("failed to save k8s CVE detail: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func ParseVulnDBData(db CVE) (*VulnDB, error) {
+	var fullVulnerabilities []*osv.OSV
+	for _, item := range db.Items {
+		if strings.Contains(excludeNonCoreComponentsCves, item.ID) {
+			continue
+		}
+		for _, cveID := range getMultiIDs(item.ID) {
+			vulnerability, err := parseMitreCve(item.ExternalURL, cveID)
+			if err != nil {
+				return nil, err
+			}
+			if cveMissingImportantData(vulnerability) {
+				continue
+			}
+			descComponent := getComponentFromDescription(item.ContentText, vulnerability.Package)
+			fullVulnerabilities = append(fullVulnerabilities, &osv.OSV{
+				ID:         cveID,
+				Modified:   item.DatePublished,
+				Published:  item.DatePublished,
+				Summary:    item.Summary,
+				Details:    vulnerability.Description,
+				Affected:   getAffectedEvents(vulnerability.versions, getComponentName(descComponent, vulnerability.Package), vulnerability.CvssV3),
+				References: []osv.Reference{{Url: item.URL}, {Url: item.ExternalURL}},
+			})
+		}
+		err := validateCvesData(fullVulnerabilities)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &VulnDB{fullVulnerabilities}, nil
+}
+
+func getAffectedEvents(v []*Version, p string, cvss Cvssv3) []osv.Affected {
+	affected := make([]osv.Affected, 0)
+	for _, av := range v {
+		if len(av.Introduced) == 0 {
+			continue
+		}
+		if av.Introduced == "0.0.0" {
+			av.Introduced = "0"
+		}
+		events := make([]osv.Event, 0)
+		ranges := make([]osv.Range, 0)
+		if len(av.Introduced) > 0 {
+			events = append(events, osv.Event{Introduced: av.Introduced})
+		}
+		if len(av.Fixed) > 0 {
+			events = append(events, osv.Event{Fixed: av.Fixed})
+		} else if len(av.LastAffected) > 0 {
+			events = append(events, osv.Event{LastAffected: av.LastAffected})
+		} else if len(av.Introduced) > 0 && len(av.LastAffected) == 0 && len(av.Fixed) == 0 {
+			events = append(events, osv.Event{LastAffected: av.Introduced})
+		}
+		ranges = append(ranges, osv.Range{
+			Events: events,
+		})
+		affected = append(affected, osv.Affected{Ranges: ranges, Package: osv.Package{Name: p, Ecosystem: "kubernetes"}, Severities: []osv.Severity{{Type: cvss.Vector, Score: fmt.Sprintf("%.1f", cvss.Score)}}})
+	}
+	return affected
+}
+
+func getComponentName(k8sComponent string, mitreComponent string) string {
+	if len(k8sComponent) == 0 {
+		k8sComponent = mitreComponent
+	}
+	if strings.ToLower(mitreComponent) != "kubernetes" {
+		k8sComponent = mitreComponent
+	}
+	return strings.ToLower(fmt.Sprintf("%s/%s", upstreamOrgByName(k8sComponent), upstreamRepoByName(k8sComponent)))
+}
+
+func validateCvesData(cves []*osv.OSV) error {
+	var result error
+	for _, cve := range cves {
+		if len(cve.ID) == 0 {
+			result = multierror.Append(result, fmt.Errorf("\nid is mssing on cve #%s", cve.ID))
```

Comment:
We no longer need `multierror` because `errors.Join` was introduced in Go 1.20. You can merge [the main branch](https://github.com/aquasecurity/vuln-list-update/commit/e9a5af7af0730c64cfab6aef705ce37c89e1db80), then Go 1.20 will be available.
https://pkg.go.dev/errors#Join